samples/dnn/dasiamrpn_tracker.cpp

```diff
@@ -0,0 +1,501 @@
+    int anchorNum = ratios.size();
+    Mat anchors;
+    Mat windows;
+    Scalar avgChans;
+    Size imgSize = {0,0};
+    Box targetBox = {0,0,0,0};
+    int scoreSize = (instanceSize-exemplarSize)/totalStride+1;
+
+    void update_scoreSize()
+    {
+        scoreSize = int((instanceSize - exemplarSize) / totalStride + 1);
+    }
+}trackerConfig;
+
+void softmax(const Mat &src, Mat & dst);
+void elementMax(Mat & src);
+Mat generateHanningWindow(const trackerConfig &trackState);
+Mat generateAnchors(trackerConfig &trackState);
+Mat getSubwindow(Mat &img, const Box &targetBox, float originalSize, Scalar avgChans);
+float trackerEval(Mat img, trackerConfig &trackState, Net &siamRPN);
+void trackerInit(Mat img, trackerConfig &trackState, Net &siamRPN, Net &siamKernelR1, Net &siamKernelCL1);
+void onMouseRectPicking(int event, int x, int y, int, void*);
+
+template<typename T>
+T sizeCal(const T &w, const T &h)
+{
+    T pad = (w + h)*0.5;
+    T sz2 = (w + pad) * (h + pad);
+    return sqrt(sz2);
+}
+
+template<>
+Mat sizeCal(const Mat &w, const Mat &h)
+{
+    Mat pad = (w + h)*0.5;
+    Mat sz2 = (w + pad).mul((h + pad));
+
+    cv::sqrt(sz2, sz2);
+    return sz2;
+}
+
+int main(int argc, char **argv)
+{
+    // Parse command line arguments.
+    CommandLineParser parser(argc, argv, keys);
+
+   if (argc == 1 || parser.has("help"))
+   {
+       parser.printMessage();
+       return 0;
+   }
+
+    std::string input = parser.get<String>("input");
+    std::string net = parser.get<String>("net");
+    std::string kernel_cls1 = parser.get<String>("kernel_cls1");
+    std::string kernel_r1 = parser.get<String>("kernel_r1");
+
+    // Read nets.
+    Net siamRPN, siamKernelCL1, siamKernelR1;
+    try
+    {
+        siamRPN = readNet(net);
+        siamKernelCL1 = readNet(kernel_cls1);
+        siamKernelR1 = readNet(kernel_r1);
+    }
+    catch (cv::Exception &ee)
+    {
+        std::cerr << "Exception: " << ee.what() << std::endl;
+        std::cout << "Can't load the network by using the flowing files:" << std::endl;
+        std::cout << "siamRPN : " << net << std::endl;
+        std::cout << "siamKernelCL1 : " << kernel_cls1 << std::endl;
+        std::cout << "siamKernelR1 : " << kernel_r1 << std::endl;
+        return 1;
+    }
+
+    const std::string preWinName = "DaSiamRPN";
+    namedWindow(preWinName, WINDOW_AUTOSIZE);
+
+    // Open a video file or an image file or a camera stream.
+    VideoCapture cap;
+    bool openSuccess = parser.has("input") ? cap.open(parser.get<String>("input")) : cap.open(0);
+    CV_Assert(openSuccess);
+
+    // Read the first image.
+    cap >> image;
+    if (image.empty())
+    {
+        std::cout << "Video reading error!" << net << std::endl;
+        return -1;
+    }
+
+    std::string infor = "Select initial bounding box you want to track.";
+    std::string infor2 = " After that, Press the ENTER key.";
+    std::cout<<infor<<std::endl;
+    std::cout<<infor2<<std::endl;
```

Comment:
no need to declare infor/infor2 variables here.

Reply:
Thank you for point out this. Fixed.